src/infrastructure/MetadataHttp.ts

```diff
@@ -0,0 +1,267 @@
+                });
+            }),
+            catchError((error) =>  throwError(error)),
+        );
+    }
+
+    /**
+     * Returns the mosaic metadata given a mosaic id and metadata key.
+     * @param namespaceId - Namespace identifier.
+     * @param key - Metadata key.
+     * @returns Observable<Metadata[]>
+     */
+    public getNamespaceMetadataByKey(namespaceId: NamespaceId, key: string): Observable<Metadata[]> {
+        return observableFrom(
+            this.metadataRoutesApi.getNamespaceMetadataByKey(namespaceId.toHex(), key)).pipe(
+            map((response: { response: ClientResponse; body: MetadataEntriesDTO; }) => {
+                const metadataEntriesDTO = response.body;
+                return metadataEntriesDTO.metadataEntries.map((metadataEntry) => {
+                    return this.buildMetadata(metadataEntry);
+                });
+            }),
+            catchError((error) =>  throwError(error)),
+        );
+    }
+
+    /**
+     * Returns the namespace metadata given a mosaic id and metadata key.
+     * @param namespaceId - Namespace identifier.
+     * @param key - Metadata key.
+     * @param publicKey - Sender public key
+     * @returns Observable<Metadata>
+     */
+    public getNamespaceMetadataByKeyAndSender(namespaceId: NamespaceId, key: string, publicKey: string): Observable<Metadata> {
+        return observableFrom(
+            this.metadataRoutesApi.getNamespaceMetadataByKeyAndSender(namespaceId.toHex(), key, publicKey)).pipe(
+            map((response: { response: ClientResponse; body: MetadataDTO; }) => {
+                const metadataDTO = response.body;
+                return this.buildMetadata(metadataDTO);
+            }),
+            catchError((error) =>  throwError(error)),
+        );
+    }
+
+    /**
+     * Returns the mosaic metadata given a mosaic id.
+     * @param namespaceId - Namespace identifier.
+     * @param queryParams - Optional query parameters
+     * @returns Observable<Metadata[]>
+     */
+    private buildMetadata(metadata: MetadataDTO): Metadata {
+        const metadataEntry = metadata.metadataEntry;
+        let targetId;
+
+        switch (metadataEntry.metadataType.valueOf()) {
+            case MetadataType.Mosaic:
+                targetId = new MosaicId(metadataEntry.targetId);
+                break;
+            case MetadataType.Namespace:
+                targetId = NamespaceId.createFromEncoded(metadataEntry.targetId);
+                break;
+            default:
+                targetId = undefined;
+        }
+        return new Metadata(
+            metadata.id,
+            new MetadataEntry(
+                metadataEntry.compositeHash,
+                metadataEntry.senderPublicKey,
+                metadataEntry.targetPublicKey,
+                UInt64.fromHex(metadataEntry.scopedMetadataKey),
+                metadataEntry.metadataType.valueOf(),
+                metadataEntry.valueSize,
+                metadataEntry.value,
```

Comment:
When a metadata transaction is announced, the user uses plain text. The sdk hides all the seralization details. That's great! 
But when getting the metadata entry from the endpoint, the value is in hex. The user is charged for converting the hex back to the provided value. The value hex -> plain text conversion should be done here and not in the client like the e2e tests.